webapp/src/serviceworker.ts

```diff
@@ -96,6 +96,7 @@ function initWebappServiceWorker() {
 
         // ai
         `/blb/ai.0.js`,
+        `/blb/ai.2.min.js`,
```

Comment:
@riknoll Can you confirm if it's safe to append rather than replace the new file here? I wasn't 100% sure based on the code in the rest of the file.

Reply:
hmmm.... why not replace it? is the old file still being served? i think the serviceworker install might fail if one of the requests 404s, so if the old file is no longer present then it's entry should be removed

Reply:
^^ I wasn't sure on this either -- isn't this serviceworker shared between all tabs? How does it work if we open beta in one tab and live in another, but beta needs `ai.0.js` and live needs `ai.2.min.js`?

Reply:
Oh, I think I see, is the way it behaves that https://github.com/microsoft/pxt/blob/master/webapp/src/appcache.ts#L29 which runs from `DOMContentLoaded`  will always end up causing it to use the last loaded page's serviceworker being the one that is 'active' as it'll overwrite if the files are different, and then we use the ref (beta / index / etc) to cache the files here https://github.com/microsoft/pxt/blob/master/webapp/src/serviceworker.ts#L182, so they won't stomp all over each other even if file set is diff.

Makes sense for handling sequential loads, I'm a little iffy on the behavior if you load them both 'simultaneously' (/index in one page and /beta in a separate tab while the first one is still loading) but that's not a real scenario.

Reply:
Hmm okay, I'll switch this over to replace rather than append, then.